llm-finetuning/huggingface/hf_deployment_service.py

```diff
@@ -0,0 +1,209 @@
+"""Implementation of the Huggingface Deployment service."""
+from zenml.logger import get_logger
+from typing import Generator, Tuple, Optional, Any, List
+from zenml.services import ServiceType, ServiceState, ServiceStatus
+from zenml.services.service import BaseDeploymentService, ServiceConfig
+from huggingface_hub import (
+    InferenceClient,
+    InferenceEndpointError,
+    InferenceEndpoint,
+)
+from huggingface_hub.utils import HfHubHTTPError
+from huggingface_hub import (
+    create_inference_endpoint,
+    get_inference_endpoint,
+    list_inference_endpoints,
+)
+from huggingface.hf_deployment_base_config import HuggingFaceBaseConfig
+
+from pydantic import Field
+
+logger = get_logger(__name__)
+
+POLLING_TIMEOUT = 1200
```

Comment:
should this be a parameter somewhere for the user to control?

Reply:
This is tricky to pass to. We want some way to pass `timeout` to `provision` metthod where `POLLING_TIMEOUT` is being used. `service.start(timeout=timeout)` calls `provision` method underneath but does not provide a way to pass `timeout`.

https://github.com/zenml-io/zenml/blob/42baca0115eca3293f98468fdd7025ce20d02262/src/zenml/services/service.py#L376-L369

We can increase pass the same timeout constant `POLLING_TIMEOUT` when we call `start` function here. Right now it uses `DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT` as default value.

https://github.com/dudeperf3ct/zenml-projects/blob/feature/zencoder-huggingface-model-deployer/llm-finetuning/huggingface/hf_model_deployer.py#L93